pkg/util/log/log.go

```diff
@@ -108,40 +108,31 @@ func addLogToBuffer(logHandle func()) {
 	logsBuffer = append(logsBuffer, logHandle)
 }
 
+// This function should be called with `sw.l` held
 func (sw *DatadogLogger) replaceInnerLogger(l seelog.LoggerInterface) seelog.LoggerInterface {
-	sw.l.Lock()
-	defer sw.l.Unlock()
 
 	old := sw.inner
 	sw.inner = l
 
 	return old
 }
 
+// This function should be called with `sw.l` held
 func (sw *DatadogLogger) changeLogLevel(level string) error {
-	sw.l.Lock()
-	defer sw.l.Unlock()
-
 	lvl, ok := seelog.LogLevelFromString(strings.ToLower(level))
 	if !ok {
 		return errors.New("bad log level")
 	}
-	logger.Load().level = lvl
+	sw.level = lvl
 	return nil
 }
 
+// This function should be called with `sw.l` held
 func (sw *DatadogLogger) shouldLog(level seelog.LogLevel) bool {
-	sw.l.RLock()
-	shouldLog := level >= sw.level
-	sw.l.RUnlock()
-
-	return shouldLog
+	return level >= sw.level
 }
 
 func (sw *DatadogLogger) registerAdditionalLogger(n string, l seelog.LoggerInterface) error {
-	sw.l.Lock()
```

Comment:
**❓ question:** should we add `// This function should be called with `sw.l` held` here ?‏

Reply:
I put lock logic in line in this PR https://github.com/DataDog/datadog-agent/pull/23564, so lock will come back in this function in the next PR, for now the lock is in the function that call this one in order to avoid data race, I'll be careful in the PR mentioned above to had this comment when needed.